### PR TITLE
Visitor must be logged in to create article.

### PR DESCRIPTION
**Pull request by athifw**

### CODE REVIEW

File: app/views/articles/new.html.erb [hunk 1]
```diff
@@ -5,9 +5,7 @@
   <%= f.label :url %>
   <%= f.text_field :url %>
   <br>
-  <%= f.label :author %>
-  <%= f.text_field :author %>
-  <br>
+  <% @article.author = current_user %>
```
Comment (yaychris) [inline app/views/articles/new.html.erb]: This line isn't necessary.

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class ArticlesController < ApplicationController
+  before_action :require_login, only: [:new]
```
Comment (yaychris) [inline app/controllers/articles_controller.rb]: This should happen for the create action, too.

**Comment by yaychris** [comment]
Two small comments, otherwise :+1:

File: spec/features/articles_spec.rb [hunk 3]
```diff
@@ -15,8 +15,14 @@
   end
   
   context "creating an article" do
+    let!(:user) { create :user, email: 'article@test.com' }
     before do
       visit '/articles'
+      click_link 'Log in'
+      fill_in 'Email', with: 'article@test.com'
+      fill_in 'Password', with: 'password'
+      click_button 'Log in'
```
Comment (efatsi) [inline spec/features/articles_spec.rb:24]: As your specs get more built out, this login code will be something you want to re-use. It's fine as it is now, but keep an eye on it and look for ways in the future to refactor this into a reusable method.

**Comment by efatsi** [comment]
Looking good :+1:

**Pull request merged**